File: lang/nodejs/src/fun/vac/util/SequenceBuilder.js
"use strict"

const RandomFactory = require("fun/vac/util/RandomFactory")
const Sequences     = require("fun/vac/util/Sequences")

exports.packIncreasing = (arr) => {
    RandomFactory.launch()
    arr[0] = RandomFactory.integerN(1, 4)
    for (let i = 1, length = arr.length; i < length; i++) {
        arr[i] = arr[i - 1] + RandomFactory.integerN(1, 4)
    }
}

exports.packRandom = (arr) => {
    RandomFactory.launch()
    for (let i = 0, length = arr.length; i < length; i++) {
        arr[i] = RandomFactory.generateInteger()
    }
}

exports.packDecreasing = (arr) => {
    this.packIncreasing(arr)
    Sequences.reverse(arr)
}

File: lang/python/src/fun/vac/util/SequenceBuilder.py
from fun.vac.util import RandomFactory
from fun.vac.util import Sequences

def packIncreasing(arr: list) -> None:
    RandomFactory.launch()
    arr[0] = RandomFactory.integerN(1, 4)
    for i in range(1, len(arr)):
        arr[i] = arr[i - 1] + RandomFactory.integerN(1, 4)

def packRandom(arr: list) -> None:
    RandomFactory.launch()
    for i in range(len(arr)):
        arr[i] = RandomFactory.generateInteger()

def packDecreasing(arr: list) -> None:
    packIncreasing(arr)
    Sequences.reverse(arr)
